Qadir__Work/JavaScript/Assignment21/Question2/crudOperations.js

```diff
@@ -0,0 +1,125 @@
+        // let indexValue = rowNumberList.indexOf(rowNumber[0]);
+        // if (indexValue > -1) {
+        //     // Remove Uncheck RowId Number from list
+        //     rowNumberList.splice(indexValue, 1);
+        // }
+        // Remove Duplicate Row Id
+        rowNumberList.filter((value, index) => rowNumberList.indexOf(value) !== index);
+    }
+    else {
+        rowNumberList.push(rowNumber[0]);
+    }
+    // console.log(rowNumberList);
+}
+
+function editData(rowNumberListData) {
+    let totalColumn = table.rows[0].cells.length;
+
+    try {
+        if (rowNumberListData.length) {
+            for (let totalSelect = 0; totalSelect < rowNumberListData.length; totalSelect++) {
+                for (let colNumber = 3; colNumber <= totalColumn; colNumber++) {
+                    let cellTdId = document.getElementById(`tdRow${rowNumberListData[totalSelect]}${colNumber}`);
+                    let cellTdData = cellTdId.innerText;
+                    cellTdId.innerHTML = "";
+
+                    let txtField = document.createElement('input');
+                    txtField.type = "text";
+                    txtField.id = `textField${rowNumberListData[totalSelect]}${colNumber}`;
+                    txtField.value = cellTdData;
+                    cellTdId.appendChild(txtField);
+                }
+            }
+        }
+        else {
+            throw new Error("Please Select Row for Edit")
+        }
+    }
+    catch(error) {
+        alert(error.message);
+    }
+}
+
+function saveData(rowNumberListData) {
+    let totalColumn = table.rows[0].cells.length;
+
+    try {
+        if (rowNumberListData.length) {
+            for (let totalSelect = 0; totalSelect < rowNumberListData.length; totalSelect++) { 
+                for (let colNumber = 3; colNumber <= totalColumn; colNumber++) {
+                    let textFieldValue = document.getElementById(`textField${rowNumberListData[totalSelect]}${colNumber}`).value;
+                    let cellTdId = document.getElementById(`tdRow${rowNumberListData[totalSelect]}${colNumber}`);
+                    cellTdId.innerHTML = textFieldValue;
+                }
+                document.getElementById(`tdCb${rowNumberListData[totalSelect]}1`).checked = false;
+            }
+        }
+        else {
+            throw new Error("Please Select Row for Save")
+        }
+    }
+    catch(error) {
+        alert(error.message);
+    }
+    rowNumberList = [];
+}
+
+function deleteData(rowNumberListData) {
+    try {
+        if (rowNumberListData.length) {
+            for (let totalSelect = 0; totalSelect < rowNumberListData.length; totalSelect++) { 
+                document.getElementById(`row${rowNumberListData[totalSelect]}`).outerHTML = "";
+            }
+        }
+        else {
+            throw new Error("Please Select Row for Delete")
+        }
+    }
+    catch(error) {
+        alert(error.message);
+    }
+}
```

Comment:
operations do not work as expected, after the records are deleted

Reply:
Yes Mam.